src/mbed_tools/build/flash.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Flash binary onto the connected device."""
+
+import shutil
+import os
+from os.path import basename, exists
+from pathlib import Path
+
+from mbed_tools.devices.devices import get_connected_devices
+from mbed_tools.build.exceptions import BinaryFileNotFoundError, BinaryFlashingFailedError, DeviceNotFoundError
+
+
+def flash_dev(
+    disk: Path, image_path: Path, copy_method: str = "default", port: str = None, program_cycle_s: int = 4
+) -> None:
+    """Flash device using copy method.
+
+    Args:
+    disk: Device mount point.
+    image_path: Image file to be copied to device.
+    copy_method: Switch -c <copy_method> (default: shell).
```

Comment:
We decided the retain the API signature `flash_dev` as in the `mbed-host-tests`. Therefore, we have unused parameters here.

Reply:
Why did we decide to keep the function signature the same as the old tool?

Reply:
It was done to be in compatible with legacy modules, so that if we decide in future to switch to that then very minimal changes will be required. However, the optional parameters does have default values in legacy modules. Therefore removing optional parameters.